docs/docs/protocol-specs/gas-and-fees/tx-setup-and-teardown.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Transaction Setup and Teardown
+---
+
+# Transaction Setup and Teardown
+
+All transactions on the Aztec network have a private component, which is processed locally, and optionally have a public component, which is processed by sequencers using the [Public VM (AVM)](../public-vm/intro.md).
+
+Transactions are broken into distinct phases:
+
+1. Private setup
+2. Private app logic
+3. Public setup
+4. Public app logic
+5. Public teardown
+6. Base rollup
+
+The private setup phase is used to specify what public function will be called for public teardown, and what entity will pay the transaction fee (i.e. the `fee_payer`).
+
+The "setup" phases are "non-revertible", meaning that if execution fails, the transaction is considered invalid and cannot be included in a block.
+
+If execution fails in the private app logic phase, the user will not be able to generate a valid proof of their private computation, so the transaction will not be included in a block.
+
+If the execution fails in the public app logic the _side effects_ from private app logic and public app logic will be reverted, but the transaction can still be included in a block. Execution then proceeds to the public teardown phase.
+
+If the execution fails in the public teardown phase, the _side effects_ from private app logic, public app logic, and public teardown will be reverted, but the transaction can still be included in a block. Execution then proceeds to the base rollup phase.
+
+In the event of a failure in public app logic or teardown, the user is charged their full [gas limit](./specifying-gas-fee-info.md#gaslimits-and-teardowngaslimits) for the transaction across all dimensions.
```

Comment:
I'm still not sure about this. Doesn't it create a dangerous incentive for sequencers to influence transactions to fail (by e.g. including their own transactions earlier in the block) so they can bank all of the fees for incurring little expense?

Reply:
I added a note to the public app logic section of the kernel tracking page.